cpp/src/arrow/compute/light_array_test.cc

```diff
@@ -216,27 +215,6 @@ TEST(KeyColumnArray, SliceBool) {
   }
 }
 
-TEST(KeyColumnArray, FromExecBatch) {
```

Comment:
Hmm, it's a bit odd to have these tests split this way.  Do you want to just move this entire test file to `exec`?  Maybe later we can, in a follow-up, attempt to pull out the utilities that are needed in `exec/test_util.h` so the entire test file can move out of `exec`.

However, for now, I think it's probably ok that this is tested in exec.

Reply:
In the coming CR I'm moving a few more things around so I'll change this to a single file as well.